tests/test_paperqa.py

```diff
@@ -948,126 +984,114 @@ def test_langchain_embeddings():
     )
 
 
-class TestVectorStore(IsolatedAsyncioTestCase):
-    async def test_langchain_vector_store(self):
-        from langchain_community.vectorstores.faiss import FAISS
-        from langchain_openai import OpenAIEmbeddings
-
-        some_texts = [
-            Text(
-                embedding=OpenAIEmbeddings().embed_query("test"),
-                text="this is a test",
-                name="test",
-                doc=Doc(docname="test", citation="test", dockey="test"),
-            )
-        ]
+@pytest.mark.asyncio()
+async def test_langchain_vector_store():
+    from langchain_community.vectorstores.faiss import FAISS
+    from langchain_openai import OpenAIEmbeddings
 
-        # checks on builder
-        try:
-            index = LangchainVectorStore()
-            index.add_texts_and_embeddings(some_texts)
-            raise "Failed to check for builder"  # type: ignore[misc]  # noqa: B016
-        except ValueError:
-            pass
-
-        try:
-            index = LangchainVectorStore(store_builder=lambda x: None)  # noqa: ARG005
-            raise "Failed to count arguments"  # type: ignore[misc]  # noqa: B016
-        except ValueError:
-            pass
-
-        try:
-            index = LangchainVectorStore(store_builder="foo")
-            raise "Failed to check if builder is callable"  # type: ignore[misc]  # noqa: B016
-        except ValueError:
-            pass
-
-        # now with real builder
-        index = LangchainVectorStore(
-            store_builder=lambda x, y: FAISS.from_embeddings(x, OpenAIEmbeddings(), y)
+    some_texts = [
+        Text(
+            embedding=OpenAIEmbeddings().embed_query("test"),
+            text="this is a test",
+            name="test",
+            doc=Doc(docname="test", citation="test", dockey="test"),
         )
-        assert index._store is None
-        index.add_texts_and_embeddings(some_texts)
-        assert index._store is not None
-        # check search returns Text obj
-        data, score = await index.similarity_search(None, "test", k=1)  # type: ignore[unreachable]
-        print(data)
-        assert type(data[0]) == Text
-
-        # now try with convenience
-        index = LangchainVectorStore(cls=FAISS, embedding_model=OpenAIEmbeddings())
-        assert index._store is None
+    ]
+
+    index = LangchainVectorStore()
+    with pytest.raises(ValueError):  # noqa: PT011
         index.add_texts_and_embeddings(some_texts)
-        assert index._store is not None
 
-        docs = Docs(
-            texts_index=LangchainVectorStore(
-                cls=FAISS, embedding_model=OpenAIEmbeddings()
-            )
-        )
-        assert docs._embedding_client is not None  # from docs_index default
+    with pytest.raises(ValueError):  # noqa: PT011
```

Comment:
FYI you can do `pytest.raises(ValueError, match="some aspect of message")` for even more precise assertion